executor/analyze.go

```diff
@@ -965,44 +980,28 @@ func (e *AnalyzeFastExec) handleSampTasks(bo *tikv.Backoffer, workID int, err *e
 	if e.ctx.GetSessionVars().GetReplicaRead().IsFollowerRead() {
 		snapshot.SetOption(kv.ReplicaRead, kv.ReplicaReadFollower)
 	}
-	rander := rand.New(rand.NewSource(e.randSeed + int64(workID)))
 
 	for i := workID; i < len(e.sampTasks); i += e.concurrency {
 		task := e.sampTasks[i]
 		if task.SampSize == 0 {
 			continue
 		}
 
-		var tableID int64
-		var minRowID, maxRowID kv.Handle
-		startKey, endKey := task.Location.StartKey, task.Location.EndKey
-		tableID, minRowID, *err = tablecodec.DecodeRecordKey(startKey)
+		step := uint32((task.EndOffset - task.BeginOffset) / task.SampSize)
+		snapshot.SetOption(kv.SampleStep, step)
```

Comment:
We also need to set following options to minimize the cost on KV.
```
NotFillCache = true
IsolationLevel = kv.RC
Priority = kv.PriorityLow
```

Reply:
Well, `tikvSnapshot` doesn't have the `IsolationLevel` field but we can add one and use it to build scan request.

Reply:
`pb.ScanRequest` does not have a field like `IsolationLevel`. How to wrap this message into the request?